docs/notebooks/documents/getting_started.py

```diff
@@ -0,0 +1,159 @@
+# for the contract object (see the PDO documentation for more information on contexts).  The context
+# includes all information necessary to create the contract object (and its dependencies) based on
+# the initial set of parameters provided by the factory notebook. Beyond the initial parameters, the
+# context allows for fine-grained customization of the contract object, though the defaults are
+# usually sufficient.
+#
+# *Operate on the Contract*: The contract object may be used once it has been created and
+# initialized. In general, cells in this section of the notebook are turned off by default; that is,
+# `%%skip True` is added at the top of the cell. To perform an operation, change the top line to
+# `%%skip False` and evaluate the cell.
+#
+# *View Contract Metadata*: For the curious (and those debugging contract behavior), the final
+# section of most notebooks includes a section to examine the metadata associated with a contract.
+
+# %% [markdown]
+# ## Configure the PDO Environment
+#
+# The remainder of this notebook helps to setup and verify the PDO environment in which the
+# notebooks operate. If you have started the Jupyter Lab server through the docker test interface, a
+# basic environment has already been created that is sufficient for working with the notebooks.
+# Otherwise, you may use the sections of this notebook to create or import signing keys for
+# transactions, configure PDO services (e.g. by importing a PDO `site.toml` file), and configure
+# service groups to simplify contract creation.
+#
+# %% [markdown]
+# ## Initialize the Interpreter
+#
+# The following cell loads modules that configure the Jupyter interpreter and simplify interaction
+# with the PDO environment.
+# %%
+import os
+import pdo.contracts.jupyter as pc_jupyter
+import IPython.display as ip_display
+import ipywidgets
+
+pc_jupyter.load_ipython_extension(get_ipython())
+
+try : state
+except NameError:
+    (state, bindings) = pc_jupyter.initialize_environment('unknown')
+
+# %% [markdown]
+# ## Create Keys
+#
+# PDO uses ECDSA cryptographic keys to sign transactions. Each key pair represents an identity or
+# role. The list below will show a list of currently available keys. There is a form to use to
+# create new keys (the refresh button will update the key list when you create new keys).
+# Additional management of keys is provided by the [Key Manager Notebook](key_manager.ipynb).
+#
+# %%
+from pdo.contracts.keys import PrivateKeyListWidget, GenerateKeyWidget
+
+key_list = PrivateKeyListWidget(state, bindings)
+key_gen = GenerateKeyWidget(state, bindings)
+ip_display.display(ipywidgets.VBox([key_list, key_gen]))
+
+# %% [markdown]
+# ## Import Site Information
+#
+# PDO clients require information about the services that will be used for contracts. Typically, a
+# service provider will create a site description file (often called `site.toml`) that can be used
```

Comment:
What are the assumptions about where PDO clients can obtain site.toml files to import?

Reply:
we have none per se. for the moment, my preference is to keep that "out of band".

Reply:
Alright, that's fine. I always like to make assumptions like these explicit (if they aren't already somewhere) to avoid any confusion.